esm_runscripts/prev_run.py

```diff
@@ -0,0 +1,493 @@
+        standard ``<dict>.get`` would do.
+        """
+
+        key = args[0]
+        # If the previous config is not loaded yet (no file foudn), try to load it
+        if not self._prev_config and len(self._config) > 0:
+            self.prev_run_config()
+        # If the previous config was loaded, use get
+        if self._prev_config:
+            if key == "NONE_YET":
+                value = {}
+            else:
+                value = self._prev_config.get(*args, **kwargs)
+        # If the previous config is not loaded yet, return get of an empty dict
+        else:
+           value = {}.get(*args, **kwargs)
+
+        return value
+
+
+    def prev_run_config(self):
+        """
+        If all the necessary information is available, loads the previous config file
+        for each component. The component loop is only run once per model. Once loaded,
+        the ``get`` and ``__get__`` methods will not call this method anymore.
+        """
+        # Logic for interactive questions about the config file
+        fromdir = os.path.realpath(self._config["general"]["started_from"])
+        scriptsdir = os.path.realpath(
+            f"{self._config['general']['experiment_dir']}/scripts/"
+        )
+        # This is necessary to display the message only once, instead of twice
+        self.warn = (
+            fromdir == scriptsdir and
+            self._config["general"].get("jobtype", "") == "compute"
+        )
+        # Check for interactive, or submitted from a computing node, to avoid
+        # using ``input()`` or ``questionaries`` in the second case
+        self.isinteractive = (
+            self._config["general"].get("last_jobtype", "") == "command_line"
+        )
+
+        # Loop through components
+        components = self._components
+        for component in components:
+            # Check if the ``prev_run_config_file`` was previously found already. If
+            # not, try to find it
+            if self._prcf.get(component):
+                prev_run_config_file, calc_prev_date = self._prcf[component]
+            else:
+                prev_run_config_file, calc_prev_date = self.find_config(component)
+                self._prcf[component] = (prev_run_config_file, calc_prev_date)
+
+            # If the file exists, load the file content
+            if os.path.isfile(prev_run_config_file):
+                # For debugging purposes: this line should not be reached more that
+                # once per component including ``prev_run`` variables
+                self._prev_config_count += 1
+                #print(f"PREV CONFIG COUNT: {self._prev_config_count}")
+
+                # Open the file and load the previous run information
+                with open(prev_run_config_file, "r") as prev_file:
+                    prev_config = yaml.load(prev_file, Loader=yaml.FullLoader)
+                prev_config = prev_config["dictitems"]
```

Comment:
I fixed this `dictitem` part and will make PR about that

Reply:
Ok, it is here https://github.com/esm-tools/esm_runscripts/pull/152
Now, you can safely remove this line since there will be no more `dictitems`, in the finished config yaml file 😀